### PR TITLE
Connecticut 2022 temporary child tax rebate

### LINKED ISSUE #4579: Connecticut 2022 temporary child tax rebate
[$750 ($250 per child up to three children)](https://portal.ct.gov/drs/credit-programs/child-tax-rebate/overview)

### PR DESCRIPTION
**Pull request by JudyZ13**
Fixes #4579

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: we need to add this to the CT net income tree
It should be a non-refundable credit 
Is it only for 2022 or also for 2023 / 2024 /2025?

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+description: Connecticut restricts child tax rebate up to this amount.
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut child tax rebate maximum
+  reference:
+    - title: 2022 Child Tax Rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml:7]: can we find the legal code source?

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+description: Connecticut restricts child tax rebate up to this amount.
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut child tax rebate maximum
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml:5]:
```suggestion
  label: Connecticut child tax rebate cap
```

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml:1]: rename to cap.yaml

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+description: Connecticut restricts child tax rebate up to this amount.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/max.yaml:1]:
```suggestion
description: Connecticut caps the child tax rebate to this amount.
```

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligibility.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from policyengine_us.model_api import *
+
+
+class ct_child_tax_rebate_eligibility(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligibility.py:4]:
```suggestion
class ct_child_tax_rebate_eligible(Variable):
```

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligibility.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    value_type = bool
+    entity = TaxUnit
+    label = "Connecticut child tax rebate eligibility"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligibility.py:7]:
```suggestion
    label = "Eligible for the Connecticut child tax rebate"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+description: Connecticut provides tax rabate at this amount per child
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut child tax rebate amount
+  reference:
+    - title: 2022 Child Tax Rebate
+      href: https://portal.ct.gov/drs/credit-programs/child-tax-rebate/overview
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml:8]: Could you look through the pdf and see if it mentions it:
https://cga.ct.gov/2022/ACT/PA/PDF/2022PA-00118-R00HB-05506-PA.PDF

File: policyengine_us/parameters/gov/states/ct/tax/income/credits/non_refundable.yaml [hunk 7]
```diff
@@ -7,3 +7,4 @@ metadata:
 values:
   2021-01-01:
     - ct_property_tax_credit
+    - ct_child_tax_rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/credits/non_refundable.yaml:10]: cant add it to 2021 

Need to check whether only applied in 2022

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/limit.yaml:1]: rename to income_limit.yaml

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/limit.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+description: Connecticut sets the child tax rebate income limits based on the filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/limit.yaml:1]:
```suggestion
description: Connecticut limits the child tax rebate to filers with income below this thresholds, based on filing status.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+description: Connecticut provides tax rabate at this amount per child
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml:1]:
```suggestion
description: Connecticut provides a child tax rebate of this amount, per child.
```

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    def formula(tax_unit, period, parameters):
+        taxable_income = tax_unit("ct_taxable_income", period)
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.states.ct.tax.income.rebate
+
+        income_eligibility = p.limit[filing_status] >= taxable_income
+
+        dependent_eligibility = (
+            tax_unit("tax_unit_count_dependents", period) > 0
+        )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py:20]:
```suggestion
```

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    label = "Eligible for the Connecticut child tax rebate"
+    definition_period = YEAR
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        taxable_income = tax_unit("ct_taxable_income", period)
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.states.ct.tax.income.rebate
+
+        income_eligibility = p.limit[filing_status] >= taxable_income
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py:16]:
```suggestion
        return p.limit[filing_status] >= taxable_income
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml [hunk 12]
```diff
@@ -0,0 +1,14 @@
+description: Connecticut caps the child tax rebate to this amount.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:1]:
```suggestion
description: Connecticut caps the child tax rebate at this amount.
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
is this in the legal code? if not please add a comment in the relevant files

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/income_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/income_limit.yaml:1]: image


I think we need a reduction folder 
With start.yaml - amount.yaml - increment.yaml

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  unit: currency-USD
+  label: Connecticut child tax rebate cap
+  reference:
+    - title: 2022 Child Tax Rebate
+      href: https://portal.ct.gov/drs/credit-programs/child-tax-rebate/overview
+    - title: TSSB 2022-5 - Frequently Asked Questions Concerning the Child Tax Rebate (4)
+      href: https://portal.ct.gov/-/media/drs/publications/tssb/2022/tssb-2022-5.pdf#page=1
+    - title: House Bill No. 5506 Public Act No. 22-118
+      href: https://cga.ct.gov/2022/ACT/PA/PDF/2022PA-00118-R00HB-05506-PA.PDF#page=548
+    #not mentioned in the legal code
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:13]:
```suggestion
    # The legal code section has been repealed since
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:13]: same comment for all yaml files

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:1]: rename to child_cap and make the value 3 as opposed to 750 - this is how the bill defines it

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut child tax rebate cap
+  reference:
+    - title: 2022 Child Tax Rebate
+      href: https://portal.ct.gov/drs/credit-programs/child-tax-rebate/overview
+    - title: TSSB 2022-5 - Frequently Asked Questions Concerning the Child Tax Rebate (4)
+      href: https://portal.ct.gov/-/media/drs/publications/tssb/2022/tssb-2022-5.pdf#page=1
+    - title: House Bill No. 5506 Public Act No. 22-118
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:11]:
```suggestion
    - title: House Bill No. 5506 Public Act No. 22-118 Sec. 411 (b) (1)
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ct/tax/income/rebate/cap.yaml:11]: specific paragraph for each parameter

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_rebate_eligible.py:1]: I dont think we need this as the rebates is phased out as oposed to a hard cut off

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/increment.yaml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut reduces the child tax rebate amount for each of these increments of federal adjusted gross income exceeding the threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/increment.yaml:1]:
```suggestion
description: Connecticut reduces the child tax rebate amount for each of these increments of state adjusted gross income exceeding the threshold.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/rate.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut reduces the personal exemption amount by this rate for each increment of federal adjusted gross income exceeding the threshold.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/rate.yaml:1]:
```suggestion
description: Connecticut reduces the personal exemption amount by this rate for each increment of state adjusted gross income exceeding the threshold.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/start.yaml [hunk 17]
```diff
@@ -0,0 +1,23 @@
+description: Connecticut reduces the child tax rebate for filers with federal adjusted gross income above this amount, based on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/reduction/start.yaml:1]:
```suggestion
description: Connecticut reduces the child tax rebate for filers with state adjusted gross income above this amount, based on filing status.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml [hunk 18]
```diff
@@ -0,0 +1,15 @@
+description: Connecticut provides a child tax rebate of this amount, per child.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/rebate/amount.yaml:1]:
```suggestion
description: Connecticut provides a child tax rebate of this amount, for each eligible child.
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_tax_rebate.py [hunk 19]
```diff
@@ -7,12 +7,23 @@ class ct_child_tax_rebate(Variable):
...
+        reduction_per_increment = p.reduction.rate * increment
+
+        excess = max_(income - reduction_start, 0)
+        increments = np.ceil(excess / increment)
+        total_reduction_amount = increments * reduction_per_increment
+
         count_dependents = tax_unit("tax_unit_count_dependents", period)
-        rebate = count_dependents * p.amount
+        child_max = p.child_cap
+        total_rebate = min_(count_dependents, child_max) * p.amount
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ct/tax/income/rebate/ct_child_tax_rebate.py:27]: delete above line
```suggestion
        capped_children = min_(count_dependents, p.child_cap)
        total_rebate = capped_children * p.amount
```

File: policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/rebate/ct_child_tax_rebate.yaml [hunk 20]
```diff
@@ -1,23 +1,29 @@
-- name: Ineligible single household with 2 dependents
+- name: Single with $101,000 agi and 2 dependents
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/rebate/ct_child_tax_rebate.yaml:1]: and below
```suggestion
- name: Single with $101,000 AGI and 2 dependents
```

**Pull request merged**